README.md

```diff
@@ -308,6 +312,41 @@ Some special commands are defined in the grammar to control the way the parser e
 
 `.import relative_filename` Allows the wbnf file to merge the grammar of the imported filename into the current grammar (equivalent to `#include` in c)
 
+`.macro Name(args) { term }` Allows the use of macros to minimise repetition in the grammar (see below)
+
+#### Macros
+
+Macros can be used when a common pattern is required through the grammar which cant easily be converted to a rule.
+An example is the required terms to support indentation:
+
+```text
+expr_block  -> arg=expr? "->" ("<" expr_ret_type ">")? "(" scopevar=NAME? ":"
+        \n+ level=INDENT expr_stmt:INDENT_SEP  \s* ")" EOL;
+rest_endpoint -> http_path attribs? ":" \n+ level=INDENT (method_def | rest_endpoint):INDENT_SEP;
+
+INDENT      -> %level="" \s+;
+INDENT_SEP  -> \n+ %level;
+
+```
+
+In the above example, the `\n+ level=INDENT <some term>:INDENT_SEP` is repeated, and unable to be pulled out to a simple 
+rule. To solve this problem macros have been implemented in wbnf. To use a macro they must be defined in the grammar
+ (does not be before first use) using the `.macro` pragma.
+
+example:
+```text
+.macro Indented(child) { \n+ level=INDENT %child:INDENT_SEP }
```

Comment:
Macro parameters shouldn't require special syntax inside the macro body. This should work:
```
.macro Indented(child) { \n+ level=INDENT child:INDENT_SEP }
```

Reply:
I'm just a bit worried that that could lead to ambiguity if the arg name is the same as any prod names?

Reply:
That's no different to parameters vs globals in regular programming languages.

Reply:
ok fair enough, I'll make the change